xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-realtime/plugin.js

```diff
@@ -69,9 +69,25 @@
           previousValue: null
         };
         editor.on('beforeSetMode', this.beforeSetMode.bind(this));
+        editor.on('mode', this.mode.bind(this));
       });
     },
 
+    mode: function(event) {
+      // The user should not be able to join the realtime editing session while in source mode.
+      // We disable the allow realtime checkbox while in source mode, and enable it when we go back to wysiwyg.
+      const editor = event.editor;
+      const realtimeCheckbox = editor._realtimeInterface.getAllowRealtimeCheckbox();
+
+      if (editor.mode === 'source') {
+        realtimeCheckbox.prop('disabled', true);
+      }
+
+      if (editor.mode === 'wysiwyg') {
+        realtimeCheckbox.prop('disabled', false);
+      }
```

Comment:
What you did is fine, but I think it can be simplified.

```suggestion
      // Realtime editing is currently supported only in WYSIWYG edit mode.
      realtimeCheckbox.prop('disabled', editor.mode !== 'wysiwyg');
```